torchgeo/datamodules/deepglobelandcover.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""DeepGlobe Land Cover Classification Challenge datamodule."""
+
+from typing import Any, Dict, Optional
+
+import pytorch_lightning as pl
+from torch.utils.data import DataLoader, Dataset
+from torchvision.transforms import Compose
+
+from ..datasets import DeepGlobeLandCover
+from .utils import dataset_split
+
+
+class DeepGlobeLandCoverDataModule(pl.LightningDataModule):
+    """LightningDataModule implementation for the DeepGlobe Land Cover dataset.
+
+    Uses the train/test splits from the dataset.
+
+    """
+
+    def __init__(
+        self,
+        root_dir: str,
+        batch_size: int = 64,
+        num_workers: int = 0,
+        val_split_pct: float = 0.2,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a LightningDataModule for DeepGlobe Land Cover based DataLoaders.
+
+        Args:
+            root_dir: The ``root`` argument to pass to the DeepGlobe Dataset classes
+            batch_size: The batch size to use in all created DataLoaders
+            num_workers: The number of workers to use in all created DataLoaders
+            val_split_pct: What percentage of the dataset to use as a validation set
+        """
+        super().__init__()  # type: ignore[no-untyped-call]
+        self.root_dir = root_dir
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.val_split_pct = val_split_pct
+
+    def preprocess(self, sample: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform a single sample from the Dataset.
+
+        Args:
+            sample: input image dictionary
+
+        Returns:
+            preprocessed sample
+        """
+        sample["image"] = sample["image"].float()
+        sample["image"] /= 255.0
+        return sample
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        """Initialize the main ``Dataset`` objects.
+
+        This method is called once per GPU per run.
+
+        Args:
+            stage: stage to set up
+        """
+        transforms = Compose([self.preprocess])
+
+        dataset = DeepGlobeLandCover(self.root_dir, "train", transforms=transforms)
+
+        self.train_dataset: Dataset[Any]
+        self.val_dataset: Dataset[Any]
+
+        if self.val_split_pct > 0.0:
+            self.train_dataset, self.val_dataset, _ = dataset_split(
+                dataset, val_pct=self.val_split_pct, test_pct=0.0
+            )
+        else:
+            self.train_dataset = dataset
+            self.val_dataset = dataset
+
+        self.test_dataset = DeepGlobeLandCover(
+            self.root_dir, "test", transforms=transforms
+        )
+
+    def train_dataloader(self) -> DataLoader[Any]:
```

Comment:
Would it be more specific to type these as `DataLoader[Dict[str, Any]]`?

Reply:
It isn't clear to me how I can find what "DataLoader" expects as a type argument though

Reply:
I believe `Dataset` and `DataLoader` type args are the return type of the dataset. But since we're using `Dataset[Any]` I don't think there's an easy way for mypy to figure out that this is correct. Might want to just ignore this.

Reply:
Trying it anyway :)